### PR TITLE
RoyaltyDomain refactor

### PR DESCRIPTION
**Pull request by Suficio**
Refactors RoyaltyDomain and CreatorsDomain to split royalty share and authorization logic.

- Documents both domains in their new form
- Accessibility improvements in creating RoyaltyDomain
- Allow NFTs to be withdrawn from Inventory to an address without going through launchpad

### CODE REVIEW

File: sources/standards/royalties/royalty.move [hunk 1]
```diff
@@ -18,25 +29,210 @@ module nft_protocol::royalty {
...
+    }
+
+    /// Remove a share attribution from an address
+    ///
+    /// Attributed addresses can only remove their own attributions. Shares of
+    /// the removed attribution are allocated to the provided address which
+    /// must already be attributed. Ensures that the total sum of shares
+    /// remains constant.
+    //
+    // TODO: Create removal methods which split shares evenly and
```
Comment (bausano) [inline sources/standards/royalties/royalty.move:205]: Please create an issue and link it here

**Review by bausano — APPROVED** [review]

File: sources/standards/royalties/royalty.move [hunk 2]
```diff
@@ -18,25 +29,210 @@ module nft_protocol::royalty {
...
+    ///
+    /// ##### Panics
+    ///
+    /// Panics if the transaction sender does not have a large enough royalty
+    /// share to transfer to the new creator or is not attributed in the first
+    /// place.
+    //
+    // TODO: Add share method for empty RoyaltyDomain controlled by
+    // CreatorsDomain
+    public fun add_share(
```
Comment (bausano) [inline sources/standards/royalties/royalty.move:169]: so the share does no longer add up to 10k

File: sources/standards/royalties/royalty.move [hunk 3]
```diff
@@ -18,25 +29,210 @@ module nft_protocol::royalty {
...
         aggregations: Bag,
+        /// Royalty share received by addresses
+        royalty_shares_bps: VecMap<address, u16>,
+    }
+
+    /// Creates an empty `RoyaltyDomain` object
+    ///
+    /// By not attributing any addresses, nobody will ever be able to claim
+    /// royalties from this `RoyaltyDomain` object.
+    public fun new_empty(ctx: &mut TxContext): RoyaltyDomain {
```
Comment (bausano) [inline sources/standards/royalties/royalty.move:82]: If someone creates and empty domain, how do they add a share to it?

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,9 @@ The format is based on [Keep a
 Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+- Refactored RoyaltyDomain and CreatorsDomain to split royalty share and authorization logic.
```
Comment (nmboavida) [inline CHANGELOG.md:12]: Changelog: Added entry function redeem_nft_transfer thus allowing creators to retrieve NFTs from their private invetories.

**Review by nmboavida — APPROVED** [review]

File: sources/standards/royalties/royalty.move (hunk 1 above)
Comment (Suficio) [reply sources/standards/royalties/royalty.move]: This behaviour is possible using existing methods, removed the TODO.

File: sources/standards/royalties/royalty.move (hunk 3 above)
Comment (Suficio) [reply sources/standards/royalties/royalty.move:82]: Added `add_share_to_empty` and `add_shares_to_empty`.

File: CHANGELOG.md (hunk 4 above)
Comment (Suficio) [reply CHANGELOG.md:13]: Done

**Pull request merged**